File: practice15/11.cpp
//https://www.nowcoder.com/practice/3fbd8fe929ea4eb3a254c0ed34ac993a?tpId=90&tqId=30782&tPage=1&rp=1&ru=/ta/2018test&qru=/ta/2018test/question-ranking
#include<iostream>
#include<string>
using namespace std;
int main()
{
	string str;
	int i, m, temp, count;
	char c1, c2;
	while (cin >> str)
	{
		m = str.size();
		temp = 0, count = 0;
		for (i = 0; i < m - 1; i++)
		{
			c1 = str[i], c2 = str[i + 1];
			if (str[i] != str[i + 1])
			{
				count++;
			}
			else
			{
				if (count > temp)
				{
					temp = count;
				}
				count = 0;
			}
		}
		if (count > temp)
		{
			temp = count;
		}
		cout << temp + 1 << endl;
	}
	system("pause");
	return 0;
}

File: practice33/11.java


//https://www.nowcoder.com/practice/1b58fbb97bc048b8a4438173c5392167?tpId=85&tqId=29882&tPage=3&rp=3&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.math.BigInteger;
import java.util.*;
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int[] num=new int[10001];
        int x;
        for(int i=0;i<9997;i++){
            x=sc.nextInt();
            num[x]=1;
        }
        StringBuilder sb=new StringBuilder();
        for(int i=1;i<=10000;i++){
            if(num[i]==0){
                sb.append(i);
            }
        }
        BigInteger number=new BigInteger(sb.toString());
        System.out.println(number.mod(BigInteger.valueOf(7)));
    }
}

